### PR TITLE
feat(unlock-app) - lock settings page

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

**Loading state** 
Screenshot 2022-11-15 at 16 54 03

**Overview** 
Screenshot 2022-11-15 at 16 54 45

Screenshot 2022-11-15 at 16 55 06

Screenshot 2022-11-15 at 16 55 13

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/locks/Settings/elements/MembershipTerms.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  lockAddress,
+  network,
+  isManager,
+  lock,
+  isLoading,
+}: MembershipTermsProps) => {
+  const settings: SettingProps[] = [
+    {
+      label: 'Duration',
+      description: 'Set up how long each membership last.',
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/elements/MembershipTerms.tsx:34]:
```suggestion
      description: 'Set up how long each membership lasts.',
```

File: unlock-app/src/components/interface/locks/Settings/elements/MembershipTerms.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          network={network}
+          duration={lock?.expirationDuration}
+          isManager={isManager}
+          disabled={!isManager}
+        />
+      ),
+    },
+    {
+      label: 'Quantity',
+      description: 'The maximum number of memberships that can be sold.',
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/elements/MembershipTerms.tsx:47]:
```suggestion
      description: 'The maximum number of memberships that can be sold. Note: There is no limit to the number of memberships that can be airdropped by a lock manager or key granter.',
```

File: unlock-app/src/components/interface/locks/Settings/elements/MembershipTerms.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+          network={network}
+          maxNumberOfKeys={lock?.maxNumberOfKeys ?? 0}
+          isManager={isManager}
+          disabled={!isManager}
+        />
+      ),
+    },
+    {
+      label: 'Price',
+      description: 'The price that you are charging for membership.',
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/elements/MembershipTerms.tsx:60]:
```suggestion
      description: 'The price that the membership contract is charging for one membership.',
```

File: unlock-app/src/components/interface/locks/Settings/elements/MembershipTerms.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+      ),
+    },
+    {
+      label: 'Recurring',
+      description:
+        'Automatically charge membership for certain amount of time.',
+      children: null,
+    },
+    {
+      label: 'Credit Card Payment',
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/elements/MembershipTerms.tsx:78]:
```suggestion
      label: 'Credit Card Payment',
```

File: unlock-app/src/components/interface/locks/Settings/elements/MembershipTerms.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+      description:
+        'Accept credit cards & ACH payments. Service & Credit card processing fees will apply to your member.',
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/elements/MembershipTerms.tsx:80]:
```suggestion
        'Accept credit cards, Apple Pay and Google Pay. Service & Credit card processing fees will be applied to the price paid by the member.',
```

File: unlock-app/src/components/interface/locks/Settings/elements/MembershipTerms.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    },
+    {
+      label: 'Transfer',
+      description: 'Allow members to transfer membership from one to others.',
+      children: null,
+    },
+    {
+      label: 'Cancellation',
+      description:
+        'Select how you would process when member cancel the membership.',
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/elements/MembershipTerms.tsx:91]:
```suggestion
        'Select how your contract should handle cancellations and optionally issue refunds.',
```

File: unlock-app/src/components/interface/locks/Settings/elements/Roles.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  return (
+    <div className="grid grid-cols-1 gap-6">
+      <SettingCard
+        label="Lock Manager"
+        description="By default, the creator of a Lock is the first Lock Manager, granting them the highest level of permissions for the lock. You can also assign this role to other wallets. Be careful: this role can’t be revoked, but only renounced."
+      ></SettingCard>
+
+      <SettingCard
+        label="Verifier"
+        description="Best use for in person event. Verifiers are trusted users at an event who can use a smart phone camera to scan a ticket QR code at the check-in to a venue and mark a ticket as checked-in."
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/elements/Roles.tsx:19]:
```suggestion
        description="Best use for in person event. Verifiers are trusted users at an event who can use a smart phone camera to scan a ticket QR code at the check-in to a venue and mark a ticket as checked-in."
```

File: unlock-app/src/components/interface/locks/Settings/forms/UpdatePriceForm.tsx [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+                step={0.01}
+                disabled={isFree || disabled}
+                {...register('keyPrice', {
+                  required: !isFree,
+                  min: 0,
+                })}
+              />
+              {errors?.keyPrice && (
+                <span className="absolute -mt-1 text-xs text-red-700">
+                  Please enter a positive number
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/forms/UpdatePriceForm.tsx:186]: Please ensure this still allows users to set `0`.

File: unlock-app/src/components/interface/locks/Settings/forms/UpdatePriceForm.tsx [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+            </div>
+          </div>
+        </div>
+
+        <span className="text-sm text-gray-600">
+          When changing the price, please remember that users who have approved
+          recurring memberships will need to manually renew in order to approve
+          the price change. Similarly if users cancel their membership, they
+          will receive a refund in the new currency, and an amount calculated
+          based on the new pricing, not the old one
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/forms/UpdatePriceForm.tsx:198]:
```suggestion
          based on the new pricing, not the old one.
```

**Review by julien51 — APPROVED** [review]
Looking good! Once you fix CI ok to merge so we can test on staging!

File: unlock-app/src/components/interface/locks/Settings/index.tsx [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+          network={network}
+          isManager={isManager}
+          lock={lock}
+          isLoading={isLoading}
+        />
+      ),
+      sidebar: (
+        <SidebarCard
+          src="/images/illustrations/img-terms.svg"
+          description="Membership Terms ipsum dolor sit amet consectetur. Magna neque facilisis eu feugiat consectetur congue."
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/index.tsx:94]:
```suggestion
          description="Membership Terms include the price, currency, duration, payment mechanisms... as well as cancellation terms and transfer fees."
```

File: unlock-app/src/components/interface/locks/Settings/index.tsx [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+        />
+      ),
+    },
+    {
+      label: 'General',
+      children: <span></span>,
+      sidebar: (
+        <SidebarCard
+          src="/images/illustrations/img-general.svg"
+          description="General ipsum dolor sit amet consectetur. Magna neque facilisis eu feugiat consectetur congue."
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/index.tsx:120]:
```suggestion
          description="Change the name and ticker for your membership contract."
```

File: unlock-app/src/components/interface/locks/Settings/elements/SettingCard.tsx [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+            />
+          </Disclosure.Button>
+          {children && (
+            <Disclosure.Panel className="pt-10">{children}</Disclosure.Panel>
+          )}
+        </div>
+      )}
+    </Disclosure>
+  )
+}
```
Comment (searchableguy) [inline unlock-app/src/components/interface/locks/Settings/elements/SettingCard.tsx:54]: We have this in metadata form as well.

Could we create a reusable component now?

File: unlock-app/src/components/interface/locks/Settings/forms/UpdateQuantityForm.tsx [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+  maxNumberOfKeys,
+  onUpdate,
+  isManager,
+  disabled,
+}: UpdateQuantityFormProps) => {
+  const [unlimitedQuantity, setUnlimitedQuantity] = useState(false)
+  const walletService = useWalletService()
+
+  useEffect(() => {
+    setUnlimitedQuantity(UNLIMITED_KEYS_COUNT == maxNumberOfKeys)
```
Comment (searchableguy) [inline unlock-app/src/components/interface/locks/Settings/forms/UpdateQuantityForm.tsx:34]: Are we depending on implicit type coercion here? Why not ===?

**Review by searchableguy — APPROVED** [review]

File: unlock-app/src/components/interface/locks/Settings/elements/SettingCard.tsx (hunk 12 above)
Comment (kalidiagne) [reply unlock-app/src/components/interface/locks/Settings/elements/SettingCard.tsx:55]: added `TODO`, ill add the component later and changed it also in metadata

**Pull request merged**